lib/http.c

```diff
@@ -1388,6 +1395,37 @@ static int http_getsock_do(struct connectdata *conn,
   return GETSOCK_WRITESOCK(0);
 }
 
+static void add_haproxy_protocol_header(struct connectdata *conn, bool *done)
```

Comment:
This better return `CURLcode` since it uses functions that can fail and those error are better propagate back up.